### PR TITLE
Add more detailed release-process instructions

### PR DESCRIPTION
**Pull request by rocky**

### CODE REVIEW

**Review by mmatera — APPROVED** [review]
@rocky , Thank you very much for this very detailed guide! 
LGTM

File: docs/project-administration/releasing/mathics-core.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+Check package from github
+=========================
+
+TODO: turn this into a script in ``admin-tools``
+
+.. code::
+
+    $ [[ ! -d /tmp/gittest ]] && mkdir /tmp/gittest; pushd /tmp/gittest
+    $ pyenv local pyston-2.3.5 # Use a version that is not the most recent
```
Comment (mmatera) [inline docs/project-administration/releasing/mathics-core.rst:85]: Just a silly comment: what about if you use conda in your system?

Comment (rocky) [reply docs/project-administration/releasing/mathics-core.rst:85]: I don't know much about conda. How would this work?

**Comment by rocky** [comment]
> @rocky , Thank you very much for this very detailed guide! LGTM

Unfortunately, we still haven't finished the last release - natlang and omnibus still need updating (which means this will probably change some more). 

Hopefully,  things will get sorted out. this weekend. And we may have to or want to go back and do some 0.1 updates for packages.

File: docs/project-administration/releasing/mathics-core.rst (hunk 1 above)
Comment (mmatera) [reply docs/project-administration/releasing/mathics-core.rst:85]: Sure. I will look at it over the weekend.

File: docs/project-administration/releasing/mathics-core.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    $ pip install -e git+https://github.com/Mathics3/${package}.git#egg=${PACKAGE}
```
Comment (mmatera) [inline docs/project-administration/releasing/mathics-core.rst]: here ${PACKAGE} should be `mathics-core`, isn't it?

Comment (rocky) [reply docs/project-administration/releasing/mathics-core.rst]: > here ${PACKAGE} should be `mathics-core`, isn't it?

Yes, that is correct it is set on line 7. The variables `package` is probably a poor choice. Probably, `github-top-dir` would be better.  I tend to adjust the notes as I go do the release, so there is a bit of sloppiness for a while.

File: docs/project-administration/releasing/mathics-core.rst (hunk 1 above)
Comment (rocky) [reply docs/project-administration/releasing/mathics-core.rst:85]: > Sure. I will look at it over the weekend.

Feel free to just go and adjust the instructions if you find a better way or more natural way to do things.

**Pull request merged**